src/VisualStudio/Core/Def/Implementation/CodeLens/RemoteCodeLensReferencesService.cs

```diff
@@ -104,11 +97,145 @@ internal sealed class RemoteCodeLensReferencesService : ICodeLensReferencesServi
                     return await CodeLensReferencesServiceFactory.Instance.GetFullyQualifiedName(solution, documentId, syntaxNode, cancellationToken).ConfigureAwait(false);
                 }
 
-                // TODO: send telemetry on session
                 return await remoteHostClient.TryRunCodeAnalysisRemoteAsync<string>(
                     solution, nameof(IRemoteCodeLensReferencesService.GetFullyQualifiedName),
                     new object[] { documentId, syntaxNode.Span }, cancellationToken).ConfigureAwait(false);
             }
         }
+
+        private static async Task<IEnumerable<ReferenceLocationDescriptor>> FixUpDescriptors(
+            Solution solution, IEnumerable<ReferenceLocationDescriptor> descriptors, CancellationToken cancellationToken)
+        {
+            var list = new List<ReferenceLocationDescriptor>();
+            foreach (var descriptor in descriptors)
+            {
+                var referencedDocumentId = DocumentId.CreateFromSerialized(
+                    ProjectId.CreateFromSerialized(descriptor.ProjectGuid), descriptor.DocumentGuid);
+
+                var document = solution.GetDocument(referencedDocumentId);
+
+                var spanMapper = document?.Services.GetService<ISpanMappingService>();
+                if (spanMapper == null)
+                {
+                    // for normal document, just add one as they are
+                    list.Add(descriptor);
+                    continue;
+                }
+
+                var span = new TextSpan(descriptor.SpanStart, descriptor.SpanLength);
+                var results = await spanMapper.MapSpansAsync(document, SpecializedCollections.SingletonEnumerable(span), cancellationToken).ConfigureAwait(false);
+                if (results.IsDefault)
+                {
+                    // handle external mapper not behavior correctly
+                    list.Add(descriptor);
+                    continue;
+                }
+
+                var result = results[0];
+                if (result.IsDefault)
+                {
+                    // it is allowed for mapper to return default 
```

Comment:
result is not span. it is MappedSpanResult. so for 0 lenght span, it can still return MappedSpanResult.Span which length 0.

if you are asking what happens if return results is 0. (not default, but 0). our contract is order/count of input should be preserved. so it should not return 0 since we gave in 1 input.